runtimes/peregrine/src/runtime_apis.rs

```diff
@@ -506,37 +506,34 @@ impl_runtime_apis! {
 
 			let currency_supply = BondedFungibles::total_issuance(currency_id.to_owned());
 
-			let (low, high, _) = match operation {
+			let (low, high, rounding) = match operation {
 				Operation::Mint(amount) => (currency_supply, currency_supply.saturating_add(amount), Round::Up),
 				Operation::Burn(amount) => (currency_supply.saturating_sub(amount), currency_supply, Round::Down),
 			};
 
-			Self::quote_for_low_and_high_bounds(pool_id, currency_idx, low, high)
+			Self::quote_for_low_and_high_bounds(pool_id, currency_idx, low, high, rounding)
 		}
 
 		fn quote_for_low_and_high_bounds(
 			pool_id: AccountId,
 			currency_idx: u8,
 			low: Balance,
 			high: Balance,
+			rounding: Round,
 		) -> Result<Balance, BondedCurrencyError> {
 			let pool = Pools::<Runtime>::get(pool_id).ok_or(BondedCurrencyError::PoolNotFound)?;
 			let PoolDetailsOf::<Runtime> { curve, bonded_currencies, denomination, collateral, .. } = pool;
-			let currency_id = bonded_currencies.get(currency_idx.saturated_into::<usize>()).ok_or(BondedCurrencyError::CurrencyNotFound)?;
-
 			let collateral_denomination = NativeAndForeignAssets::decimals(collateral);
 
-			let normalized_low = balance_to_fixed(low, denomination, Round::Down).map_err(|_| BondedCurrencyError::BalanceConversion)?;
-			let normalized_high = balance_to_fixed(high, denomination, Round::Up).map_err(|_| BondedCurrencyError::BalanceConversion)?;
+			let normalized_low = balance_to_fixed(low, denomination, rounding).map_err(|_| BondedCurrencyError::BalanceConversion)?;
+			let normalized_high = balance_to_fixed(high, denomination, rounding).map_err(|_| BondedCurrencyError::BalanceConversion)?;
+
 
-			let passive_supply = bonded_currencies
-				.iter()
-				.filter_map(|id| (id != currency_id).then_some(BondedFungibles::total_issuance(id.to_owned())))
-				.map(|supply| balance_to_fixed(supply, denomination, Round::Up).map_err(|_| BondedCurrencyError::BalanceConversion))
-				.collect::<Result<Vec<FixedPoint>, BondedCurrencyError>>()?;
+			let (_, passive_supply) = BondedCurrencies::calculate_normalized_passive_issuance(&bonded_currencies, denomination, currency_idx.into(), rounding)
```

Comment:
a u8 can be infallibly converted to a usize - saturated_into() shouldn't be necessary